### PR TITLE
refactor and update old/deprecated components

### PR DESCRIPTION
**Pull request by Zacharis278**
@edx/masters-devs-cosmonauts 

~I'll be putting together a sandbox for further manual testing but I believe the code itself is ready for review.~
This code can be tested here: https://learner-portal-test-learner-portal.sandbox.edx.org/datascience
The `verified` learner has enrollments already set up for this program.

1. Replaces old/deprecated frontend components with frontend-platform versions.
2. We need to use a copy of the most recent header component since it doesn't support custom drop downs (yet)
3. Cleaned up dead code paths relating to the enterprise portal

Repositories to be deprecated by this pull request:
https://github.com/edx/frontend-learner-portal-base
https://github.com/edx/frontend-component-site-header
https://github.com/edx/frontend-component-site-footer

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -20,7 +20,7 @@ UNBRANDED_LANDING_PAGE=''
 USE_PROGRAMS_MOCK_DATA=''
 IDP_SLUG='saml-default'
 # uncomment the line below to use mock designer data from @edx/gatsby-source-portal-designer
-# USE_PROGRAMS_MOCK_DATA=true
+USE_PROGRAMS_MOCK_DATA=true
```
Comment (Zacharis278) [inline .env.development:23]: reminder: remove this

File: package.json [hunk 2]
```diff
@@ -6,13 +6,9 @@
...
     "@edx/brand": "npm:@edx/brand-openedx@^1.1.0",
-    "@edx/frontend-analytics": "^2.0.0",
-    "@edx/frontend-auth": "^5.3.4",
-    "@edx/frontend-component-footer": "^3.0.4",
-    "@edx/frontend-component-site-header": "^2.5.0",
-    "@edx/frontend-i18n": "^2.1.1",
-    "@edx/frontend-learner-portal-base": "^1.0.7",
-    "@edx/frontend-logging": "^2.0.1",
+    "@edx/frontend-component-footer-edx": "^3.1.0",
+    "@edx/frontend-component-header": "^2.2.1",
```
Comment (bseverino) [inline package.json:10]: Is this still needed?

**Review by bseverino — APPROVED** [review]
LGTM pending sandbox tests. Thank you for doing this!

File: package.json (hunk 2 above)
Comment (Zacharis278) [reply package.json:10]: nope, definitely not

File: src/assets/icons/menu.svg [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<svg width="24px" height="24px" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
+    <title>menu</title>
+    <desc>Created with Sketch.</desc>
+    <g id="menu" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <rect id="Rectangle" fill="currentColor" x="2" y="5" width="20" height="2"></rect>
+        <rect id="Rectangle" fill="currentColor" x="2" y="11" width="20" height="2"></rect>
+        <rect id="Rectangle" fill="currentColor" x="2" y="17" width="20" height="2"></rect>
+    </g>
+</svg>
```
Comment (schenedx) [inline src/assets/icons/menu.svg:11]: #NIT new line please.

File: src/components/layout/Layout.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
```
[the remark this answers is not in the document]
Comment (Zacharis278) [reply src/components/layout/Layout.jsx:1]: the layout component used to exist in the base repo

**Pull request merged**